src/mqttgw.erl

```diff
@@ -1564,10 +1605,11 @@ delete_client_dynsubs(Subject, BrokerConn, BrokerId, UniqueId, SessionPairId, Ti
     DynSubL = mqttgw_dynsub:list(Subject),
 
     %% Send a multicast event to the application
+    %% But only about apps/* subs
     [send_dynsub_multicast_event(
         <<"subscription.delete">>, Subject, Data, BrokerConn, BrokerId,
         UniqueId, SessionPairId, Time)
-     || Data <- DynSubL],
+     || Data <- DynSubL, hd(maps:get(topic, Data, [""])) == <<"apps">>],
```

Comment:
Why do you use a string here `[""]`?

Reply:
both `hd()` and `lists:nth()` will throw if the list is empty, `[""]` allows to avoid this case by always providing at least one element

Reply:
Yes. Why the string?